### PR TITLE
Preparations for `no-std` support

### PR DESCRIPTION
**Pull request by str4d**
Part of #211.

### CODE REVIEW

File: src/constants/sinsemilla.rs [hunk 1]
```diff
@@ -161,27 +161,18 @@ mod tests {
     #[test]
     fn i2lebsp_k_round_trip() {
         {
-            let bitstring = (0..K).map(|_| rand::random()).collect::<Vec<_>>();
-            assert_eq!(
-                i2lebsp_k(lebs2ip_k(&bitstring) as usize).to_vec(),
-                bitstring
-            );
+            let bitstring = [0; K].map(|_| rand::random());
```
Comment (nuttycom) [inline src/constants/sinsemilla.rs:164]: You can map arrays now??? :exploding_head:

File: src/bundle.rs [hunk 2]
```diff
@@ -10,142 +10,34 @@ use nonempty::NonEmpty;
...
-    cmx: ExtractedNoteCommitment,
-    /// The transmitted note ciphertext.
-    encrypted_note: TransmittedNoteCiphertext,
-    /// A commitment to the net value created or consumed by this action.
-    cv_net: ValueCommitment,
-    /// The authorization for this action.
-    authorization: A,
-}
-
 impl<T> Action<T> {
```
Comment (nuttycom) [inline src/bundle.rs:25]: Why is this still allowed to be here without violating the Orphan rule?

**Review by nuttycom — APPROVED** [review]
utACK

File: src/bundle.rs (hunk 2 above)
Comment (str4d) [reply src/bundle.rs:25]: The orphan rule applies to crates, not modules. We're still inside the `orchard` crate; it doesn't matter where in it we impl these. See also how we implemented `zcash_address`.

File: src/constants/sinsemilla.rs (hunk 1 above)
Comment (str4d) [reply src/constants/sinsemilla.rs:164]: Yep, since 1.55.

**Comment by str4d** [comment]
[Force-pushed](https://github.com/zcash/orchard/compare/34ad90bc4636e523c9b10f76b80eaabeef6a00cb..5c5c9994393df1b982957b756f72f31f0abea4df) to fix a broken intra-doc link.

**Pull request merged**